pkg/controller/scyllacluster/resource.go

```diff
@@ -648,17 +650,46 @@ func stringOrDefault(str, def string) string {
 	return def
 }
 
-func MakeServiceAccount(sc *scyllav1.ScyllaCluster) *corev1.ServiceAccount {
+func MakeServiceAccount(sc *scyllav1.ScyllaCluster, serviceAccounts map[string]*corev1.ServiceAccount, serviceAccountLister corev1listers.ServiceAccountLister) (*corev1.ServiceAccount, error) {
+	annotations := map[string]string{}
+	name := naming.MemberServiceAccountNameForScyllaCluster(sc.Name)
+
+	existing, found := serviceAccounts[name]
+	if !found {
+		// In 1.7.0 we started managing ServiceAccounts. Because previously these were created by users
+		// there's a high chance that they don't match our selector, hence are not available in serviceAccounts.
+		// We need to copy annotations stored in user SA because these may control integrations with k8s provider services
+		// like IAM management.
+		// Ref: https://github.com/scylladb/scylla-operator/issues/932
+		//
+		// Cache fetch happens only once - when SA is being adopted, or it's missing.
+		// It can be removed in 1.8.0 because all SAs will be already adopted, and they will match our selector.
+		// TODO: https://github.com/scylladb/scylla-operator/issues/934
+		var err error
+		existing, err = serviceAccountLister.ServiceAccounts(sc.Namespace).Get(name)
+		if err != nil && !apierrors.IsNotFound(err) {
+			return nil, fmt.Errorf("can't get %s service account: %w", naming.ManualRef(sc.Namespace, name), err)
+		}
+		found = err == nil
```

Comment:
reusing the var seems a bit confusing because on the first look these 2 ifs could be an else block, but it's just a reused variable